research-app-messages/src/selections.ts

```diff
@@ -8,9 +62,9 @@ export interface Source {
   ra: number;
   dec: number;
   name: string;
-  catalogName: string;
+  layer: LayerInfo;
   zoomDeg?: number;
-  catalogData: {
+  layerData: {
```

Comment:
Note that by removing the existing `catalogName` and `catalogData` fields, you are breaking the messaging API here. In this  case, I think that's OK since we have strong reason to believe that there is zero code that actually depends on the existing API so far. But in other cases it's important to preserve existing fields/behaviors/etc for compatibility.

Also note that the docstring will need updating since it references `catalogName`.

See also some comments on terminology that I'll have below.